File: sprint3/By languages/JS/lesson9code1.js
let digit_lengths = [4, 4, 3, 3, 6, 4, 5, 4, 6, 6]; // длины слов «ноль», «один»,...

function key_for_card(card) {
    return [-digit_lengths[card], card];
}

let cards = [2, 3, 7];
cards.sort((a, b) => key_for_card(a) < key_for_card(b) ? -1 : 1);
console.log(cards);

File: sprint3/By languages/go/lesson9code1.go
package main

import (
    "fmt"
    "sort"
)

var digitLengths = []int{4, 4, 3, 3, 6, 4, 5, 4, 6, 6} // длины слов «ноль», «один»,...

func keyForCard(card int) []int {
    return []int{-digitLengths[card], card}
}

func main() {
    cards := []int{2, 3, 7}
    sort.Slice(cards, func(i, j int) bool {
        return keyForCard(cards[i])[0] < keyForCard(cards[j])[0]
    })
    fmt.Println(cards)
}
